util/backup/advanced/restore-redis-infisical.sh

```
@@ -0,0 +1,590 @@
+    *)
+        echo "ERROR: Invalid environment '$ENVIRONMENT'. Valid environments: dev, stg, prd"
+        echo ""
+        show_usage
+        exit 1
+        ;;
+esac
+
+# =============================================================================
+# INFISICAL DELEGATION
+# =============================================================================
+export BACKUP_ENCRYPTION_KEY=$(infisical secrets get BACKUP_ENCRYPTION_KEY --env="$ENVIRONMENT" --plain)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)